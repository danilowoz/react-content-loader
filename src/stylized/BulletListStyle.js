//@flow
import * as React from 'react'
import Wrap from '../Wrap'
import type { WrapProps } from '../Wrap'

const BulletListStyle = (props: WrapProps): React.Element<*> => (
  <Wrap {...props}>
    <circle cx="10" cy="20" r="8" />
    <rect x="25" y="15" rx="5" ry="5" width="220" height="10" />
    <circle cx="10" cy="50" r="8" />
    <rect x="25" y="45" rx="5" ry="5" width="220" height="10" />
    <circle cx="10" cy="80" r="8" />
    <rect x="25" y="75" rx="5" ry="5" width="220" height="10" />
    <circle cx="10" cy="110" r="8" />
    <rect x="25" y="105" rx="5" ry="5" width="220" height="10" />
  </Wrap>
)

export default BulletListStyle
