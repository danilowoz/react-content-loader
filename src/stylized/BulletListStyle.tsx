import * as React from 'react'

import Holder from '../Holder'
import { IContentLoaderProps } from '../'

const ReactContentLoaderBulletList: React.FC<IContentLoaderProps> = props => (
  <Holder {...props}>
    <circle cx="10" cy="20" r="8" />
    <rect x="25" y="15" rx="5" ry="5" width="220" height="10" />
    <circle cx="10" cy="50" r="8" />
    <rect x="25" y="45" rx="5" ry="5" width="220" height="10" />
    <circle cx="10" cy="80" r="8" />
    <rect x="25" y="75" rx="5" ry="5" width="220" height="10" />
    <circle cx="10" cy="110" r="8" />
    <rect x="25" y="105" rx="5" ry="5" width="220" height="10" />
  </Holder>
)

export default ReactContentLoaderBulletList
