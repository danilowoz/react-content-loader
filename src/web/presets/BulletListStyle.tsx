import * as React from 'react'

import { IContentLoaderProps } from '..'
import ContentLoader from '../ContentLoader'

const ReactContentLoaderBulletList: React.FC<IContentLoaderProps> = props => (
  <ContentLoader viewBox="0 0 245 125" {...props}>
    <circle cx="10" cy="20" r="8" />
    <rect x="25" y="15" rx="5" ry="5" width="220" height="10" />
    <circle cx="10" cy="50" r="8" />
    <rect x="25" y="45" rx="5" ry="5" width="220" height="10" />
    <circle cx="10" cy="80" r="8" />
    <rect x="25" y="75" rx="5" ry="5" width="220" height="10" />
    <circle cx="10" cy="110" r="8" />
    <rect x="25" y="105" rx="5" ry="5" width="220" height="10" />
  </ContentLoader>
)

export default ReactContentLoaderBulletList
