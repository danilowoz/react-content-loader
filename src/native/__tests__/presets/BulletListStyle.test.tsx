import 'react-native'

import * as React from 'react'
import * as renderer from 'react-test-renderer'

import BulletListStyle from '../../presets/BulletListStyle'

describe('BulletListStyle', () => {
  const wrapper = renderer.create(
    <BulletListStyle uniqueKey="BulletListStyle" animate={false} speed={20} />
  )

  test('renders correctly', () => {
    const tree = wrapper.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('props are propagated ', () => {
    expect(wrapper.root.props.speed).toEqual(20)
  })
})
