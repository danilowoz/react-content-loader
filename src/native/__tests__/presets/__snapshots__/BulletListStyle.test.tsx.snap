// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulletListStyle renders correctly 1`] = `
<Svg
  height={125}
  style={Object {}}
  viewBox="0 0 245 125"
  width={245}
>
  <Rect
    clipPath="url(#BulletListStyle-animated-diff)"
    fill="url(#BulletListStyle-diff)"
    height="100%"
    width="100%"
    x="0"
    y="0"
  />
  <Defs>
    <ClipPath
      id="BulletListStyle-animated-diff"
    >
      <Circle
        cx="10"
        cy="20"
        r="8"
      />
      <Rect
        height="10"
        rx="5"
        ry="5"
        width="220"
        x="25"
        y="15"
      />
      <Circle
        cx="10"
        cy="50"
        r="8"
      />
      <Rect
        height="10"
        rx="5"
        ry="5"
        width="220"
        x="25"
        y="45"
      />
      <Circle
        cx="10"
        cy="80"
        r="8"
      />
      <Rect
        height="10"
        rx="5"
        ry="5"
        width="220"
        x="25"
        y="75"
      />
      <Circle
        cx="10"
        cy="110"
        r="8"
      />
      <Rect
        height="10"
        rx="5"
        ry="5"
        width="220"
        x="25"
        y="105"
      />
    </ClipPath>
    <LinearGradient
      id="BulletListStyle-diff"
      x1="-100%"
      x2="100%"
      y1={0}
      y2={0}
    >
      <Stop
        offset={0}
        stopColor="#f5f6f7"
      />
      <Stop
        offset={0}
        stopColor="#eee"
      />
      <Stop
        offset={0}
        stopColor="#f5f6f7"
      />
    </LinearGradient>
  </Defs>
</Svg>
`;
